src/main/java/gov/usgs/earthquake/geoserve/ANSSRegionsFactory.java

```diff
@@ -0,0 +1,197 @@
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import gov.usgs.earthquake.geoserve.GeoserveLayersService;
+import gov.usgs.earthquake.qdm.Regions;
+import gov.usgs.util.FileUtils;
+import gov.usgs.util.StreamUtils;
+import gov.usgs.util.XmlUtils;
+
+/**
+ * Class to manage ANSS Authoritative Region updates.
+ *
+ * Simplest usage:
+ *     ANSSRegionsFactory.getFactory().getRegions()
+ * 
+ * Regions are not fetched until {@link #startup()}
+ * (or {@link #fetchRegions()}) is called.
+ */
+public class ANSSRegionsFactory {
+
+    // logging object
+    public static final Logger LOGGER = Logger.getLogger(ANSSRegionsFactory.class.getName());
+
+    // milliseconds per day
+    public static final long MILLISECONDS_PER_DAY = 86400000L;
+
+    // path to write regions.json
+    public static final String REGIONS_JSON = "regions.json";
+
+    // global factory object
+    private static ANSSRegionsFactory SINGLETON;
+
+    // service used to load regions
+    private GeoserveLayersService geoserveLayersService;
+
+    // the current regions object.
+    private Regions regions;
+
+    // timer used to auto fetch region updates
+    private Timer updateTimer = new Timer();
+
+
+    /**
+     * Use default GeoserveLayersService.
+     */
+    public ANSSRegionsFactory () {
+        this(new GeoserveLayersService());
+    }
+
+    /**
+     * Use custom GeoserveLayersService.
+     */
+    public ANSSRegionsFactory (final GeoserveLayersService geoserveLayersService) {
+        this.geoserveLayersService = geoserveLayersService;
+    }
+
+    /**
+     * Get the global ANSSRegionsFactory, 
+     * creating and starting if needed.
+     */
+    public static ANSSRegionsFactory getFactory() {
+        if (SINGLETON == null) {
+            SINGLETON = new ANSSRegionsFactory();
+            SINGLETON.startup();
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                SINGLETON.shutdown();
+            }));
+        }
+        return SINGLETON;
+    }
+
+    /**
+     * Set the global ANSSRegionsFactory,
+     * shutting down any existing factory if needed.
+     */
+    public static void setFactory(final ANSSRegionsFactory factory) {
+        if (SINGLETON != null) {
+            SINGLETON.shutdown();
+        }
+        SINGLETON = factory;
+    }
+
+    /**
+     * Get the service.
+     */
+    public GeoserveLayersService getGeoserveLayersService() {
+        return this.geoserveLayersService;
+    }
+
+    /**
+     * Set the service.
+     */
+    public void setGeoserveLayersService(final GeoserveLayersService service) {
+        this.geoserveLayersService = service;
+    }
+
+    /**
+     * Get the most recently fetched Regions.
+     */
+    public Regions getRegions () {
+        return regions;
```

Comment:
Be consistent in referencing member variables. Is it `regions` or `this.regions`? I prefer the latter since it seems more explicit/clear.

Reply:
Made this `this.regions`